### ISSUE TITLE
`http::Request::send` panics in Node with `wasm-pack`

### ISSUE THREAD

**Issue by littlebenlittle** [rustwasm/gloo#484]
### Describe the Bug

Calls to `http::Request::send` reach the panic condition [here](https://github.com/rustwasm/gloo/blob/dc0e61e7b24fe0f33d66960385909dc0f4195bd9/crates/net/src/http/request.rs#L336) in NodeJS environment.

### Steps to Reproduce

Full crate: 
[gloo_node.tar.gz](https://github.com/user-attachments/files/16273511/gloo_node.tar.gz)

Run:

```sh
wasm-pack build --target nodejs
node index.js
```

Node Version: v21.6.1

Code:

<details>

```rs
extern crate console_error_panic_hook;
extern crate wasm_bindgen;

use std::panic;
use wasm_bindgen::prelude::*;

#[wasm_bindgen]
pub fn main() {
    panic::set_hook(Box::new(console_error_panic_hook::hook));
    wasm_bindgen_futures::spawn_local(async move {
        gloo_net::http::Request::get("localhost:8080")
            .send() // panics
            .await
            .unwrap();
    });
}
```

</details>

### Additional Info

Seems to be the result of undefined `WorkerGlobalScope`. Node v21+ defines `fetch` on the global context ([ref](https://nodejs.org/api/globals.html#fetch)) so something like this might work?

<details>

```rs
pub async fn send(self) -> Result<Response, Error> {
        let request = self.0;
        let global = js_sys::global();
        let maybe_window =
            Reflect::get(&global, &JsValue::from_str("Window")).map_err(js_to_error)?;
        let promise = if !maybe_window.is_undefined() {
            let window = global.dyn_into::<web_sys::Window>().unwrap();
            window.fetch_with_request(&request)
        } else {
            let maybe_worker = Reflect::get(&global, &JsValue::from_str("WorkerGlobalScope"))
                .map_err(js_to_error)?;
            if !maybe_worker.is_undefined() {
                let worker = global.dyn_into::<web_sys::WorkerGlobalScope>().unwrap();
                worker.fetch_with_request(&request)
            } else {
                // panic!("Unsupported JavaScript global context");
                // instead:
                let maybe_fetch = js_sys::Reflect::get(&global, &JsValue::from_str("fetch")).unwrap();
                if maybe_fetch.is_undefined() {
                    panic!("no global fetch()")
                }
                let fetch = maybe_fetch.dyn_into::<js_sys::Function>().unwrap();
                fetch.call1(&global, &unsafe { JsValue::from_abi(request.return_abi()) });
        }
        // ...
}
```

</details>

**Comment by littlebenlittle** [comment]
This is not caught by the current test suite as run [here](https://github.com/rustwasm/gloo/blob/dc0e61e7b24fe0f33d66960385909dc0f4195bd9/.github/workflows/tests.yml#L260).

It is caught when run with `wasm-pack test --node crates/net --all-features` and commenting out [this line](https://github.com/rustwasm/gloo/blob/dc0e61e7b24fe0f33d66960385909dc0f4195bd9/crates/net/tests/http.rs#L6) to disable non-browser environment.

**Comment by kunjee17** [comment]
I am getting similar kind of error for console package as well. Alert works without any issue but as soon as I add and use console or net package. it panics and doesn't work with my react project. 

```
LinkError: WebAssembly.instantiate(): Import #4 "wbg" "__wbg_log_7c3433e130418e14": function import requires a callable
```

It shows this kind of cryptic message. Any help would be really appreciated.

**Comment by littlebenlittle** [comment]
@kunjee17 Without more context I can only guess, but that looks like your wasm module declares a function import and the JS environment is giving it something that is not a function. I recommend opening a separate issue and providing a full example.

**Comment by kunjee17** [comment]
hey, @littlebenlittle thanks for replying. Here is demo [repo](https://github.com/kunjee17/moon-test ). I am basically trying to combine `moonrepo` that has `server` written in rust. Shared logic written in rust and `client` that will expose part of logic to UI written in `react`. I know everything can be written in rust. But basically I am trying to create a logical library in wasm, that can be consume in any UI library.

**Comment by littlebenlittle** [comment]
Resolved by #474

**Comment by kunjee17** [comment]
@littlebenlittle Don't kill me for this. But my issue solved by deleting `node_modules` and reinstalling. Don't know I should be happy about it or sad. In any case thanks for help man.
